txnats/protocol.py

```diff
@@ -202,9 +205,8 @@ def dataReceived(self, data):
 
                 if sid in self.unsubs:
                     self.unsubs[sid] -= 1
-                    if self.unsubs[sid] == 0:
-                        del self.unsubs[sid]
-                        self.dispatch(actions.UnsubMaxReached(sid, protocol=self))
+                    if self.unsubs[sid] <= 0:
+                        self.unsub(sid)
```

Comment:
Must check if sending an UNSUB when max messages have been received is redundant.